### PR TITLE
feat: update dependencies

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
I also replaced `rollup-plugin-node-builtins`& `rollup-plugin-node-globals` with `rollup-plugin-polyfill-node` as that's what the @rollup team [recommends](https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency) & the @babel team did as well in https://github.com/babel/babel/pull/11098.

---

BREAKING CHANGE: Use Jest v28

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@kentcdodds Seems like snapshots are only failing on Windows for some reason 🤔

**Comment by kentcdodds** [comment]
Hmmm... I'm not sure. Maybe windows is adding extra lines to the output for some reason?

File: src/config/prettierrc.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
+/** @type {import('prettier').Options} */
 module.exports = {
   arrowParens: 'avoid',
+  bracketSameLine: false,
```
Comment (MichaelDeBoey) [inline src/config/prettierrc.js:4]: https://prettier.io/docs/en/options.html#bracket-line

File: src/config/prettierrc.js [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
   htmlWhitespaceSensitivity: 'css',
   insertPragma: false,
-  jsxBracketSameLine: false,
   jsxSingleQuote: false,
   printWidth: 80,
   proseWrap: 'always',
   quoteProps: 'as-needed',
   requirePragma: false,
   semi: false,
+  singleAttributePerLine: false,
```
Comment (MichaelDeBoey) [inline src/config/prettierrc.js:16]: https://prettier.io/docs/en/options.html#single-attribute-per-line

File: src/config/rollup.config.js [hunk 3]
```diff
@@ -9,10 +9,9 @@ const {
 } = require('@rollup/plugin-node-resolve')
 const replace = require('@rollup/plugin-replace')
 const camelcase = require('lodash.camelcase')
-const {terser} = require('rollup-plugin-terser')
-const nodeBuiltIns = require('rollup-plugin-node-builtins')
-const nodeGlobals = require('rollup-plugin-node-globals')
 const omit = require('lodash.omit')
+const nodePolyfills = require('rollup-plugin-polyfill-node')
```
Comment (MichaelDeBoey) [inline src/config/rollup.config.js:13]: I also replaced `rollup-plugin-node-builtins`& `rollup-plugin-node-globals` with `rollup-plugin-polyfill-node` as that's what the @rollup team [recommends](https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency) & the @babel team did as well in https://github.com/babel/babel/pull/11098.

**Comment by nickmccurdy** [comment]
Would this be safe to merge (after resolving conflicts)? I'm running into a configuration issue for the [Recorder Extension](https://github.com/testing-library/testing-library-recorder-extension)'s e2e test suite because of the old Jest version, and it's blocking my next major release. Let me know if I can help with anything.

**Comment by kentcdodds** [comment]
Once we resolve conflicts and figure out what to do about the failing build then I'm happy to merge.

**Comment by nickmccurdy** [comment]
The build seems to be failing because the new version of `glob` requires using POSIX paths, but we're using native paths (so Windows is not using POSIX paths).

**Comment by nickmccurdy** [comment]
I think I fixed it. @MichaelDeBoey can you please review https://github.com/MichaelDeBoey/kcd-scripts/pull/66? It's set to merge into this PR.

**Review by kentcdodds — DISMISSED** [review]
Looks great! Thank you so much for spending time working on this!

**Comment by kentcdodds** [comment]
Uh oh. Looks like the build is actually failing so we won't get a release if I merge this 😬

**Comment by kentcdodds** [comment]
🤦‍♂️ just read the thread and now I understand what's going on 😅 Thanks for your work @nickmccurdy! Excited to get this merged 👍

**Review by kentcdodds — APPROVED** [review]
Super! Thank you 👏👏

**Pull request merged**